src/wallet/coincontrol.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <wallet/coincontrol.h>
+
+#include <util.h>
+
+void CCoinControl::SetNull()
+{
+    destChange = CNoDestination();
+    m_change_type.reset();
+    fAllowOtherInputs = false;
+    fAllowWatchOnly = false;
+    m_avoid_partial_spends = gArgs.GetBoolArg("-avoidpartialspends", DEFAULT_AVOIDPARTIALSPENDS);
+    setSelected.clear();
+    m_feerate.reset();
+    fOverrideFeeRate = false;
+    m_confirm_target.reset();
+    m_signal_bip125_rbf.reset();
+    m_fee_mode = FeeEstimateMode::UNSET;
```

Comment:
Generally, we initialize all non-static class members where they are declared. Ensure determinism by avoiding accidental use of uninitialized values. Also, static analyzers balk about this. Initializing the members in the declaration makes it easy to spot uninitialized ones.

https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#c-data-structures

Reply:
It would duplicate the initialization in the `SetNull()` method as well as in the variable declarations. I assume that's okay, then?

Reply:
Ah, sorry. I had missed that `SetNull()` is used in the gui code once.

Reply:
NP! I am unsure if c942091 should be dropped or if it's a-good to have. Keeping it unless/until someone complains.